src/asmcnc/apps/drywall_cutter_app/material_setup_popup.py

```diff
@@ -0,0 +1,516 @@
+                                                                      str(self.dwt_config.active_cutter.max_depth_per_pass))
+        self.pass_depth_warning_zero = "[color=#FF0000]" + self.l.get_str("Depth per pass must be greater than 0") \
+                                           + "[/color]"
+
+        self.cut_depth_warning_soft_limit = "[color=#FF0000]" + self.l.get_str("Max allowable cut is") + \
+                                            " Xmm[/color]".replace("X", str(self.soft_limit_total_cut_depth))
+        self.cut_depth_warning_cutter_max = "[color=#FF0000]" + self.l.get_str("Max depth of tool is") + \
+                                            " Xmm[/color]".replace("X",
+                                                                   str(self.dwt_config.active_cutter.max_depth_total))
+        self.cut_depth_warning_zero = "[color=#FF0000]" + self.l.get_str("Total cut depth must be greater than 0") \
+                                          + "[/color]"
+
+        self.update_strings()
+
+    def on_open(self):
+        self.load_active_config()
+
+    def load_active_config(self):
+        self.material_thickness.text = str(self.dwt_config.active_config.cutting_depths.material_thickness)
+        self.bottom_offset.text = str(self.dwt_config.active_config.cutting_depths.bottom_offset)
+        self.auto_pass_checkbox.active = self.dwt_config.active_config.cutting_depths.auto_pass
+        self.depth_per_pass.text = str(self.dwt_config.active_config.cutting_depths.depth_per_pass)
+
+    def update_strings(self):
+        self.title_label.text = self.l.get_str("Cutting depths")
+        self.material_thickness_label.text = self.l.get_str("Material thickness")
+        self.bottom_offset_label.text = self.l.get_str("Bottom offset")
+        self.total_cut_depth_label.text = self.l.get_str("Total cut depth")
+        self.auto_pass_label.text = self.l.get_str("Auto pass")
+        self.depth_per_pass_label.text = self.l.get_str("Depth per pass")
+        self.pass_depth_warning.text = self.pass_depth_warning_cutter_max
+        self.cut_depth_warning.text = self.cut_depth_warning_soft_limit
+
+    def on_touch(self):
+        for text_input in self.text_inputs:
+            text_input.focus = False
+            if text_input.text != '':
+                try:
+                    text_input.text = str(float(text_input.text))
+                except:
+                    pass
+            else:
+                text_input.text = str(float(0))
+
+    def on_checkbox_active(self):
+        if self.auto_pass_checkbox.active:
+            self.depth_per_pass.disabled = True
+            self.depth_per_pass.text = self.depth_per_pass.hint_text
+            self.depth_per_pass.hint_text = ''
+        else:
+            self.depth_per_pass.disabled = False
+            self.depth_per_pass.hint_text = self.depth_per_pass.text
+            self.depth_per_pass.text = ''
+
+    def update_graphic_position(self):
+        try:
```

Comment:
same as other comments on try: except, check for your error condition and then return if met